SampleProject/VOKCoreDataManagerTests/VOKThreadingTests.m

```diff
@@ -0,0 +1,37 @@
+//
+//  VOKThreadingTests.m
+//  Vokoder Sample Project
+//
+//  Created by Brock Boland on 2/26/16.
+//  Copyright © 2016 Vokal. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import <VOKCoreDataManager.h>
+
+@interface VOKThreadingTests : XCTestCase
+
+@end
+
+@implementation VOKThreadingTests
+
+- (void)testSetupOnBackgroundThreadFails
+{
+    XCTestExpectation *willFinish = [self expectationWithDescription:@"Background thread setup will finish"];
+
+    // Setup on the main thread should work
+    [[VOKCoreDataManager sharedInstance] resetCoreData];
+    XCTAssertNoThrow([[VOKCoreDataManager sharedInstance] setResource:@"VOKCoreDataModel" database:nil]);
+
+    NSOperationQueue *backgroundQueue = [[NSOperationQueue alloc] init];
+    backgroundQueue.qualityOfService = NSOperationQualityOfServiceBackground;
+    [backgroundQueue addOperationWithBlock:^{
+        [[VOKCoreDataManager sharedInstance] resetCoreData];
+        // Setup on a background thread should fail
```

Comment:
An abundance of caution and this comment on `setResource:database`:

> The main context will be initialized immediately. This method should only be called from the main queue

That was added by @seanwolter in 9954a8909d9a7f3cd2f4c1f4bf44143b02ce789f but I can't find the PR with the feedback that prompted it.